subsys/pm/pm_priv.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_PM_PM_PRIV_H_
+#define ZEPHYR_SUBSYS_PM_PM_PRIV_H_
+
+#if defined(CONFIG_PM_DEVICE) && !defined(CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE)
```

Comment:
That's a weird kconfig expression?  PM_DEVICE_RUNTIME_EXCLUSIVE is about device lifecycle callbakcs.  At first glance it's not clear at all to me why that has anything to do with the global CPU count?

Maybe put that test into code instead and rely on IS_ENABLED()?